# Whole exome sequencing data analysis scripts
This repository has scripts related to whole exome sequencing project.  

Below are the descriptions for the scripts in this repository.

___

### 01_bam_to_rpkm.sh
This script uses GATK Best practice workflow bam file to generate genewise rpkm.

### 02_SingleCell_CNV_rpkm.R
This script uses file sample.txt generated by `01_bam_to_rpkm.sh` to generate CNV heatmap.  

### 03_Simple_Oncoplot.R
This script uses matrix of multiple genomic alteration to generate an oncoprint.

### 04_Complex_Oncoprint.R
This script uses multiple genomic alteration, oncogene status and pathway information for genes to generate a complex oncoprint.

### 05_Phylogenetic_Tree.R
This script uses an excelsheet with snp information given based on presence and absence to generate a phylogenetic tree.

### 06_Mutational_Signature.R
This script uses tab separated file containing chromosome number, variant position, reference and alternate bases to generate mutational signature.

### 07_PAM50_breast_cancer_subtyping_for_GSE9893.R
This script uses GEOquery to generate PAM50 subtyping for GSE9893.  

---
For any queries please email us at radhav@um.edu.mo  

